src/main/webapp/app/lecture/lecture.component.html

```diff
@@ -86,13 +86,30 @@ <h2 id="page-heading">
     <div class="table-responsive" *ngIf="filteredLectures">
         <table class="table table-striped">
             <thead>
-                <tr>
-                    <th><span jhiTranslate="global.field.id">ID</span></th>
-                    <th><span jhiTranslate="artemisApp.lecture.title">Title</span></th>
-                    <th><span jhiTranslate="artemisApp.lecture.description">Description</span></th>
-                    <th><span jhiTranslate="artemisApp.lecture.startDate">Start Date</span></th>
-                    <th><span jhiTranslate="artemisApp.lecture.endDate">End Date</span></th>
-                    <th><span jhiTranslate="artemisApp.lecture.course">Course</span></th>
+                <tr jhiSort [(predicate)]="predicate" [(ascending)]="ascending" (sortChange)="sortRows()">
+                    <th class="d-md-table-cell" jhiSortBy="id">
+                        <span jhiTranslate="global.field.id">ID</span>
+                        <fa-icon [icon]="faSort"></fa-icon>
+                    </th>
+                    <th jhiSortBy="title">
+                        <span jhiTranslate="artemisApp.lecture.title">Title</span>
+                        <fa-icon [icon]="faSort"></fa-icon>
+                    </th>
+                    <th jhiSortBy="description">
+                        <span jhiTranslate="artemisApp.lecture.description">Description</span>
+                        <fa-icon [icon]="faSort"></fa-icon>
+                    </th>
+                    <th jhiSortBy="startDate">
+                        <span jhiTranslate="artemisApp.lecture.startDate">Start Date</span>
+                        <fa-icon [icon]="faSort"></fa-icon>
+                    </th>
+                    <th jhiSortBy="endDate">
+                        <span jhiTranslate="artemisApp.lecture.endDate">End Date</span>
+                        <fa-icon [icon]="faSort"></fa-icon>
+                    </th>
+                    <th class="d-md-table-cell">
```

Comment:
Do we even need the course here? Judging from your screencast, we are already within a course => I'd expect the course to be the same for all entries.

Make sure to check other component usages, though, before removing it.